test/suites/pvc/suite_test.go

```diff
@@ -0,0 +1,143 @@
+package integration
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/cloudprovider/aws/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/test"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+
+	"github.com/aws/karpenter/test/pkg/environment"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var env *environment.Environment
+
+func TestPVC(t *testing.T) {
+	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		var err error
+		env, err = environment.NewEnvironment(t)
+		Expect(err).ToNot(HaveOccurred())
+	})
+	RunSpecs(t, "PVC Handling")
+}
+
+var _ = BeforeEach(func() {
+	env.BeforeEach()
+})
+var _ = AfterEach(func() {
+	env.AfterEach()
+})
+
+// This test requires the EBS CSI driver to be installed
+var _ = Describe("Dynamic PVC", func() {
+	It("should run a pod with a dynamic persistent volume", func() {
+		// Ensure that the EBS driver is installed, or we can't run the test.
+		var ds appsv1.DaemonSet
+		if err := env.Client.Get(env.Context, client.ObjectKey{
```

Comment:
It would be interesting if we could ask the environment about whether or not things are installed.

`hasEBSCSIDriver, err := env.HasEBSCSIDriver()`
`hasAWSVPCCNI, err := env.HasEBSCSIDriver()`

Reply:
I'll leave that for the second use, right now there is only one test that cares.